### DISCUSSION TITLE
$user not capturing the logged on user, displaying "Administrator" instead

### DISCUSSION THREAD

**Discussion by teddimitrov** [starship/starship#4621] [Q&A]
On a domain joined Windows 11 with Powershell 7, the prompt displays "Administrator" instead of the currently logged on user. 

`$env:UserName` and `$(Get-WMIObject -class Win32_ComputerSystem | select username).username `

both display the logged-on user and domain\user respectively. 

Previously I had oh my posh and it was able to display the username. 

I am sure it is something simple but so far, I have not been able to coax starship to display the logged-on username instead of "Administrator"

Any suggestions would be appreciated. Thank you.

**Comment by chipbuster** [comment]
On Windows, the existing username is overwritten by "Administrator" if [the process is running with elevated privileges](https://github.com/starship/starship/blob/master/src/modules/username.rs#L73).

Unfortunately, I am not familiar enough with the Windows APIs to tell exactly what triggers this, but that is where I suspect it's coming from.

**Reply by davidkna** [reply]
The module replaces the username with `Administrator` on Windows if the user is running with elevated modules. This should perhaps become an option or default to something like `Administrators/$username`.

**Reply by hasancemcerit** [reply]
@chipbuster, @davidkna could we create a feat. request, or PR (if preferred) to change the behavior of `username` module on windows?

There's a workaround to get the `actual` username by adding a custom command, like below:
```
[custom.user]
command = '$env:USERNAME'
when = true
format = '[$output]($style)'
```
However, this adds significant and noticeable delay in starship prompt execution.

I am pretty sure, users of starship on windows/pwsh are always running with elevated privileges.😁
